docs/articles/documentation/recipes/basics/test-file-upload.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: docs
+title: Test File Upload
+permalink: /documentation/recipes/basics/test-file-upload.html
+---
+# Test File Upload
+
+This recipe shows how to test your application's file upload functionality with TestCafe.
+
+[Full Example Code](https://github.com/DevExpress/testcafe-examples/tree/master/examples/upload-files)
+
+Assume the following HTML:
+
+```html
+<body>
+    <form action="http://localhost:3000/upload" method="post" enctype="multipart/form-data">
+        <input id="upload-input" type="file" name="files" multiple />
+        <input type="submit" id="upload-btn" value="Upload Files"/>
+    </form>
+</body>
+```
+
+TestCafe's file upload emulation is differrent from what the real user would do to upload files.
+
+To manually upload files with this application, the user needs to perform the following steps:
+
+* Click the 'Choose File' button that opens the 'Choose file' dialog in the browser
+* Select the files
+* Begin the upload
+
+Since TestCafe cannot interact with the native dialog, it instead directly manipulates the `<input type=file>` elements' state with the `.setFilesToUpload` action.
+
+The sample page includes a `<form>` that contains:
+
+* `<input>` element that holds the selected files
+* the `submit` button that sends the files to the server.
+
+Below you can find full test code and a step-by-step explanation.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `./index.html`
+
+test('Select files to upload', async t => {
+
+    await t
+        .setFilesToUpload('#upload-input', [
+            './uploads/text-file-1.txt',
+            './uploads/text-file-2.txt'
+        ])
+        .click('#upload-btn');
+});
+```
+
+Start with an empty test.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `./index.html`
+
+test('Select files to upload', async t => {
+    // Here you will put test code.
+});
+```
+
+Use the [setFilesToUpload](../../reference/test-api/testcontroller/setfilestoupload.md) Method to populate the `<input>` with files. Then [click](../../reference/test-api/testcontroller/click.md) the `submit` button.
+
+```js
+await t
+        .setFilesToUpload('#upload-input', [
+            './uploads/text-file-1.txt',
+            './uploads/text-file-2.txt'
+        ])
+        .click('#upload-btn');
+```
+
+This sends the selected files to the server. Note that with TestCafe you don't need to click the `<input type="file">` element itself to select the files. Such a click would call a native file selection dialog, which TestCafe can't close or otherwise interact with.
```

Comment:
1) don't we need a comma after *Note that with TestCafe, ...*?
2) I'm okay with 'would' here, but the correctors would replace it with *Such a click calls...*
3) The same for ending sentences with a preposition - *which TestCafe can neither close nor interact with it otherwise.* (if I'm putting otherwise at the right place)

Reply:
> Note that with TestCafe, you don't need to click the `<input type="file">` element itself to select the files. Such a click calls a native file selection dialog, which TestCafe can't interact with or close. Such a click calls a native file selection dialog, which TestCafe can't interact with or close.